kernel/src/log_segment.rs

```diff
@@ -0,0 +1,165 @@
+//! Represents a segment of a delta log. [`LogSegment`] wraps a set of  checkpoint and commit
+//! files.
+
+use crate::expressions::column_expr;
+use std::sync::{Arc, LazyLock};
+use url::Url;
+
+use crate::{
+    actions::{get_log_schema, Metadata, Protocol, METADATA_NAME, PROTOCOL_NAME},
+    schema::SchemaRef,
+    DeltaResult, Engine, EngineData, Error, ExpressionRef, FileMeta,
+};
+use itertools::Itertools;
+
+#[derive(Debug)]
+#[cfg_attr(feature = "developer-visibility", visibility::make(pub))]
+#[cfg_attr(not(feature = "developer-visibility"), visibility::make(pub(crate)))]
+pub(crate) struct LogSegment {
+    pub log_root: Url,
+    /// Reverse order sorted commit files in the log segment
+    pub commit_files: Vec<FileMeta>,
+    /// checkpoint files in the log segment.
+    pub checkpoint_files: Vec<FileMeta>,
```

Comment:
Ah that was the goal earlier on in the PR. Test code in snapshot inspects log segments pretty heavily, so I removed it for the time being.

 I've also since come across several threads cautioning against the traditional wisdom of other languages to have a getter/setter. Seems that a getter is only useful if you want to be careful with maintaining internal consistency of a struct where mutability is involved. 

I'm still on the fence, since LogSegment is both 1) "just a pile of `FileMeta`s", but also 2) logically represents a contiguous section of the log. 2) seems to indicate to me that getters would protect the internal consistency that the `LogSegment` is indeed a contiguous section of the graph.

Any preferences?